gap/congruences/cong.gi

```diff
@@ -390,17 +424,69 @@ function(class, elm)
   return EquivalenceClassOfElementNC(cong, Representative(class) * elm);
 end);
 
+SEMIGROUPS.GenericCongLookup := function(cong)
+  local S, lookup, class, nr, elm;
+  S := Range(cong);
+  if not IsFinite(S) then
+    ErrorNoReturn("Semigroups: SEMIGROUPS.GenericCongLookup: usage,\n",
+                  "<cong> must be over a finite semigroup,");
+  fi;
+  lookup := [1 .. Size(S)];
+  for class in NonTrivialEquivalenceClasses(cong) do
+    nr := PositionCanonical(S, Representative(class));
+    for elm in class do
+      lookup[PositionCanonical(S, elm)] := nr;
+    od;
+  od;
+  return lookup;
+end;
+
 InstallMethod(EquivalenceRelationLookup,
 "for a semigroup congruence",
 [IsSemigroupCongruence],
-EquivalenceRelationCanonicalLookup);
+SEMIGROUPS.GenericCongLookup);
 
 InstallMethod(EquivalenceRelationLookup,
 "for a left semigroup congruence",
 [IsLeftSemigroupCongruence],
-EquivalenceRelationCanonicalLookup);
+SEMIGROUPS.GenericCongLookup);
 
 InstallMethod(EquivalenceRelationLookup,
 "for a right semigroup congruence",
 [IsRightSemigroupCongruence],
-EquivalenceRelationCanonicalLookup);
+SEMIGROUPS.GenericCongLookup);
+
+SEMIGROUPS.GenericCongCanonicalLookup := function(cong)
```

Comment:
Same comment for this as for `GenericCongLookup` I don't think this needs to hang around in `SEMIGROUPS`